HorizonEngine/src/HorizonEngine/Renderer/Sprite.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#ifndef HZN_SPRITE_H
+#define HZN_SPRITE_H
+
+#include "Texture.h"
+
+namespace Hzn
+{
+	/// <summary>
+	/// Represents a 2D Sprite. It is essentially a small portion of Sprite Sheet / Texture Atlas.
+	/// Sprite Sheet is just a special type 2D Texture, which has all the sprites that we want to render
+	/// in just one file (texture). It can be interpreted as a grid of cells of fixed dimensions.
+	/// The sides of rectangle that represents a sprite, are multiples of the sides of the grid cells
+	/// in the sprite sheet.
+	/// </summary>
+	class Sprite2D
+	{
+	public:
+		/// <summary>
+		/// Create a sprite from Sprite Sheet. 
+		/// </summary>
+		/// <param name="spriteSheet">Sprite Sheet from which you wish to create the sprite.</param>
+		/// <param name="sheetPosition">Position (x, y) of the sprite in the sheet. Bottom Left is (0, 0) and Top Right
+		/// is (width of the sheet, height of the sheet). </param>
+		/// <param name="cellSize">Dimensions of the grid cells of the sprite sheet.</param>
+		/// <param name="spriteSize">Dimensions of the sprite.</param>
+		/// <returns>Shared pointer to the 2D Sprite.</returns>
+		static std::shared_ptr<Sprite2D> create(
+			const std::shared_ptr<Texture2D>& spriteSheet,
+			const glm::vec2& sheetPosition,
+			const glm::vec2& cellSize,
+			const glm::vec2& spriteSize = { 1.0f, 1.0f }
+		);
```

Comment:
Why is shared_ptr used here? Is it so that ownership of the sprite created is shared and can be accessed by different parts?

Reply:
Hi,
its just that if you want to pass this sprite to a function and such, you can easily do so, without dealing with move semantics of passing unique_ptr through functions.